js/components/ReportSummary.vue

```diff
@@ -33,25 +35,104 @@ export default {
   },
 
   methods: {
+    /**
+     * Emit deleteSummary event.
+     */
     deleteSummary() {
       if (confirm('Permanently delete this summary?')) {
         this.$emit('deleteSummary', this.index);
       }
     },
 
+    /**
+     * Checks if the user has permission to delete a report summary.
+     * @return true if the user has permission to delete a report summary.
+     */
     canDelete() {
       // TODO: implement REDDEV-595
       return true;
+    },
+
+    /**
+     * Checks a value to determine if it is missing. Missing values are those
+     * that are zero in length, undefined, only whitespace, or null.
+     * @return true if the value is missing
+     */
+    missingValue(value) {
+      if (value == null || typeof value === 'undefined') {
+        return true;
+      } else {
+        const stringValue = new String(value);
+        const trimmed = stringValue.trim();
+        return trimmed.length ? false : true;
+      }
     }
   },
 
   computed: {
+    /**
+     * @return true if the summary contains an itemized count.
+     */
     isItemized() {
       return this.strategy === STRATEGY.ITEMIZED;
     },
 
+    /**
+     * Checks summaryData for missing values.
+     * @return true if summaryData contains a missing value.
+     */
+    hasMissingValue() {
+      return this.summaryData.filter(val => {
+        return this.missingValue(val);
+      }).length > 0;
+    },
+
+    /**
+     * Inserts MISSING value if empty.
+     * @return summaryData with empty values replaced by MISSING
+     */
+    summaryDataWithMissing() {
+      return this.hasMissingValue ? this.summaryData.map(summary => {
+        return this.missingValue(summary) ? MISSING : summary;
+      }) : this.summaryData;
+    },
+
+    /**
+     * Sum distinct summary data values and build an object of the form: `{ "field value": totalCount, "another value": anotherTotal }`
+     * @return object with value and corresponding count
+     */
     summaryCounts() {
-      return countBy(this.summaryData);
+      return countBy(this.summaryDataWithMissing);
+    },
+
+    /**
+     * Transform summaryCounts into an array of the form: `[{ label: 'foo', count: 42 }, ...]`
+     * @return an array of objects where each object has a label and a count.
+     */
+    mappedSummaryCounts() {
+      const summaryData = [];
+      Object.keys(this.summaryCounts).forEach(key => {
+        summaryData.push({
+          label: key,
+          count: this.summaryCounts[key]
+        });
+      });
```

Comment:
This might be more natural as an actual `map` call:

```javascript
const { summaryCounts } = this;
return Object.keys(summaryCounts)
  .map(key => ({ label: key, count: summaryCounts[key] });
```

Reply:
+1